```
D:\a\_work\1\s\src\Controls\tests\DeviceTests\HandlerTestBase.cs(205,12): error CS1501: No overload for method 'SetupWindowForTests' takes 3 arguments [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\HandlerTestBase.cs(205,12): error CS1501: No overload for method 'SetupWindowForTests' takes 3 arguments [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj]
D:\a\_work\1\s\src\Controls\tests\DeviceTests\HandlerTestBase.cs(205,12): error CS1501: No overload for method 'SetupWindowForTests' takes 3 arguments [D:\a\_work\1\s\src\Controls\tests\DeviceTests\Controls.DeviceTests.csproj]
```

--- a/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellRenderer.cs
+++ b/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellRenderer.cs
@@ -355,7 +355,6 @@ namespace Microsoft.Maui.Controls.Handlers.Compatibility
 			Element.SizeChanged -= OnElementSizeChanged;
 			((IShellController)Element).RemoveAppearanceObserver(this);
 
-			// This cast is necessary because IShellFlyoutRenderer doesn't implement IDisposable
 			if (_flyoutView is ShellFlyoutRenderer sfr)
 				sfr.Disconnect();
 			else

--- a/src/Controls/tests/DeviceTests/HandlerTestBase.Windows.cs
+++ b/src/Controls/tests/DeviceTests/HandlerTestBase.Windows.cs
@@ -24,12 +24,13 @@ namespace Microsoft.Maui.DeviceTests
 			== AccessibilityView.Content;
 
 
-		Task SetupWindowForTests<THandler>(IWindow window, Func<Task> runTests)
+		Task SetupWindowForTests<THandler>(IWindow window, Func<Task> runTests, IMauiContext mauiContext = null)
 			where THandler : class, IElementHandler
 		{
+			mauiContext ??= MauiContext;
 			return InvokeOnMainThreadAsync(async () =>
 			{
-				var applicationContext = MauiContext.MakeApplicationScope(UI.Xaml.Application.Current);
+				var applicationContext = mauiContext.MakeApplicationScope(UI.Xaml.Application.Current);
 
 				var appStub = new MauiAppNewWindowStub(window);
 				UI.Xaml.Application.Current.SetApplicationHandler(appStub, applicationContext);

--- a/src/Controls/tests/DeviceTests/HandlerTestBase.iOS.cs
+++ b/src/Controls/tests/DeviceTests/HandlerTestBase.iOS.cs
@@ -24,14 +24,15 @@ namespace Microsoft.Maui.DeviceTests
 			return platformView.AccessibilityElementsHidden;
 		}
 
-		Task SetupWindowForTests<THandler>(IWindow window, Func<Task> runTests)
+		Task SetupWindowForTests<THandler>(IWindow window, Func<Task> runTests, IMauiContext mauiContext = null)
 			where THandler : class, IElementHandler
 		{
+			mauiContext ??= MauiContext;
 			return InvokeOnMainThreadAsync(async () =>
 			{
 				try
 				{
-					_ = window.ToHandler(MauiContext);
+					_ = window.ToHandler(mauiContext);
 					await runTests.Invoke();
 				}
 				finally

